### DISCUSSION TITLE
segment.isFinal

### DISCUSSION THREAD

**Discussion by rcolepeterson** [speechly/speechly#313] [General]
segment.isFinal

seems to take a long time to fire true.
what can i do to speed this up?
Sometimes it is fast and sometimes it takes up to 53 seconds or more. Is this dues to internet?
How can i have speechly return final segments faster?

**Comment by teelisyys** [comment]
Hi @rcolepeterson ! Thank you for using Speechly!

By default the segment relies solely on ML to determine when the segment closes. Depending on the configuration, there may be cases where it is ambiguous whether a segment should finalise or not. In these cases the segment finalisation may be delayed.

If the segment boundary is usually marked with silence, you might want to try set the [`silence_triggered_segmentation`](https://docs.speechly.com/reference/config#silence_triggered_segmentation) option, available both in the `config.yaml` and our dashboard. This will cause segment change when the defined length of silence is encountered, which naturally finalises the segment in the process.

Please also note that the emitted transcripts are finalised separately, so depending on you use case, you might want to use them even before the segment has finalised.

**Reply by rcolepeterson** [reply]
THANK YOU!

**Comment by Lex-Kent** [comment]
@teelisyys I don't think this is entirely correct. I have a similar issue with some of my applications, but not others, regardless of what figure I put in the **_silence triggered segmentation_** text box on the dashboard. I asked about this in my other post: https://github.com/speechly/speechly/discussions/307 but I never received a reply.

I've written a workaround that stops the Speechly instance if the segment has not changed after a timeout of 2 or 3 seconds, but this means my intents don't get populated and I have to make do with the transcript. This is impossible to test in the dashboard because of course, that works with a mic button but I'm pretty sure it's a glitch with Speechly. It would help me enormously if it was addressed.

**Reply by teelisyys** [reply]
Thanks for the details! As I've now updated to the https://github.com/speechly/speechly/discussions/307, this is unfortunately caused by a bug unrelated to the silence triggered segmentation, and no generally applicable solution is available.
